src/librbd/ImageCtx.cc

```diff
@@ -850,7 +850,11 @@ struct C_InvalidateCache : public Context {
     on_safe = util::create_async_context_callback(*this, on_safe);
 
     assert(owner_lock.is_locked());
-    if (object_cacher != NULL) {
+
+    bool journaling = false;
+    journaling = (journal != NULL && !journal->is_journal_replaying());
+
+    if (!journaling && object_cacher != NULL) {
```

Comment:
@zhouyuan I just mean move your conditional to ```AioImageFlush``` and have it skip calling ```ImageCtx::flush``` when a user requests a flush op and journaling is enabled.  We need to keep ```ImageCtx::flush``` as is because it handles system flush cases where we truly do need to flush all IO out to disk.